### PR TITLE
Added wait_for_alert_by_state_sampler  and get_alerts_by_state for adding the option for checking alerts in any state

### PR DESCRIPTION
**Pull request by OhadRevah**
##### Short description:
Added the functions wait_for_alert_by_state_sampler and get_alerts_by_state in monitor.py,
Added the option to check for alerts with other states than only firing.

##### What this PR does / why we need it:
I've been testing alerts, and I needed a to check alerts that have 
a long evaluation time, in this cases we check for pending state, so I tried to add
that option and it was very handy while working with alerts.

### CODE REVIEW

File: ocp_utilities/monitoring.py [hunk 1]
```diff
@@ -129,41 +129,45 @@ def get_all_alerts_by_alert_name(self, alert_name):
                 alert_list.append(alert)
         return alert_list
 
-    def get_firing_alerts(self, alert_name):
+    def get_alerts_by_state(self, alert_name, state="firing"):
         """
-        get all the firing alerts from list of active alerts
+        get all the alerts from list of active alerts according the state
```
Comment (omrirh) [inline ocp_utilities/monitoring.py:134]:
```suggestion
        get all the alerts from list of active alerts according to the state
```

**Review by omrirh — APPROVED** [review]

**Review by myakove — CHANGES_REQUESTED** [review]
Please change the pr title to something more meaningful

File: ocp_utilities/monitoring.py [hunk 2]
```diff
@@ -129,41 +129,45 @@ def get_all_alerts_by_alert_name(self, alert_name):
                 alert_list.append(alert)
         return alert_list
 
-    def get_firing_alerts(self, alert_name):
```
Comment (myakove) [inline ocp_utilities/monitoring.py:132]: You renamed the function, this is a breaking change
Please make sure to mention it in the pr title

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Review by omrirh — APPROVED** [review]

**Review by jpeimer — CHANGES_REQUESTED** [review]
Maybe it would be better to add new functions without removing the old ones? 
And add a deprecation warning for the old ones?

**Comment by myakove** [comment]
> Maybe it would be better to add new functions without removing the old ones? And add a deprecation warning for the old ones?

Great idea @jpeimer, 
@OhadRevah Please implement it.

**Review by omrirh — APPROVED** [review]

File: ocp_utilities/monitoring.py [hunk 3]
```diff
@@ -217,3 +217,44 @@ def alerts(self):
         get all the active alerts
         """
         return self._get_response(query=f"{self.api_v1}/alerts")
+
+    def get_alerts_by_state(self, alert_name, state="firing"):
```
Comment (jpeimer) [inline ocp_utilities/monitoring.py:221]: Please reuse this function in **get_firing_alerts**

File: ocp_utilities/monitoring.py [hunk 4]
```diff
@@ -217,3 +217,44 @@ def alerts(self):
...
+        """
+        get all the alerts from list of active alerts according the state
+        """
+        alert_list = self.get_all_alerts_by_alert_name(alert_name=alert_name)
+        return [alert for alert in alert_list if alert["state"] == state]
+
+    def wait_for_alert_by_state_sampler(
```
Comment (jpeimer) [inline ocp_utilities/monitoring.py:228]: Please reuse this function in **wait_for_firing_alert_sampler**

**Review by jpeimer — CHANGES_REQUESTED** [review]
Thanks Ohad,

A couple of comments + 
Please update the commit message

**Review by omrirh — APPROVED** [review]

File: ocp_utilities/monitoring.py [hunk 5]
```diff
@@ -217,3 +202,44 @@ def alerts(self):
...
+            alert_name=alert_name,
+            state=state,
+        )
+        try:
+            for sample in sampler:
+                if sample:
+                    LOGGER.info(f"Found alert: {alert_name} in {state} state.")
+                    return sample
+
+        except TimeoutExpiredError:
```
Comment (myakove) [inline ocp_utilities/monitoring.py:243]: Please remove try:excpect
No need to do it only for the error log, You will have the info from `TimeoutExpiredError` error

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by omrirh — APPROVED** [review]

**Review by jpeimer — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

File: ocp_utilities/monitoring.py [hunk 6]
```diff
@@ -133,8 +133,7 @@ def get_firing_alerts(self, alert_name):
         """
```
[the remark this answers is not in the document]
Comment (OhadRevah) [reply ocp_utilities/monitoring.py:133]: We discussed about it earlier, I did it at first place but it decided that these functions will be an extension, you can see the previous comments from Meni  and Jenia.

File: ocp_utilities/monitoring.py [hunk 7]
```diff
@@ -133,8 +133,7 @@ def get_firing_alerts(self, alert_name):
...
         get all the firing alerts from list of active alerts
         """
-        alert_list = self.get_all_alerts_by_alert_name(alert_name=alert_name)
-        return [alert for alert in alert_list if alert["state"] == "firing"]
+        return self.get_alerts_by_state(alert_name=alert_name)
 
     def wait_for_firing_alert_sampler(self, alert_name, timeout=TIMEOUT_10MIN):
```
[the remark this answers is not in the document]
Comment (OhadRevah) [reply ocp_utilities/monitoring.py:138]: Same as my last comment.

**Review by oharan2 — APPROVED** [review]

**Pull request merged**